### PR TITLE
feat(atomic): use "product" instead of "result" in commerce components

### PR DESCRIPTION
**Pull request by y-lakhdar**
Adding translation for query-summary and load-more components.

Let me know if you think creating new components instead would make more sense

https://coveord.atlassian.net/browse/KIT-3348

### CODE REVIEW

**Review by alexprudhomme — COMMENTED** [review]
Storybook is not loading the component for the default case. 
image

File: packages/atomic/src/components/commerce/atomic-commerce-load-more-products/e2e/atomic-commerce-load-more-products.e2e.ts [hunk 1]
```diff
@@ -31,17 +31,17 @@ test('should display a progress value between 1 and 100', async ({
...
     await Promise.all(
       allProducts.map(async (product) => product.waitFor({state: 'attached'}))
     );
-    await expect(products.results).toHaveCount(96);
+    await expect(products.products).toHaveCount(96);
   });
 });
 
-test.describe('when theres no results', () => {
+test.describe('when theres no products', () => {
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/commerce/atomic-commerce-load-more-products/e2e/atomic-commerce-load-more-products.e2e.ts:44]:
```suggestion
test.describe('when there are no products', () => {
```

File: packages/atomic/src/components/commerce/atomic-commerce-query-summary/e2e/atomic-commerce-query-summary.e2e.ts [hunk 2]
```diff
@@ -19,33 +19,33 @@ test.describe('after searching for kayak', () => {
...
 
   test('should not display duration by default', async ({querySummary}) => {
-    const textRegex = /^Results 1-[\d,]+ of [\d,]+ for kayak$/;
+    const textRegex = /^Products 1-[\d,]+ of [\d,]+ for kayak$/;
     await expect(querySummary.text(textRegex)).toBeVisible();
   });
 });
 
-test.describe('when search yields no results', () => {
+test.describe('when search yields no produts', () => {
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/commerce/atomic-commerce-query-summary/e2e/atomic-commerce-query-summary.e2e.ts:27]:
```suggestion
test.describe('when search yields no products', () => {
```

File: packages/atomic/src/components/commerce/atomic-commerce-query-summary/e2e/atomic-commerce-query-summary.e2e.ts [hunk 3]
```diff
@@ -19,33 +19,33 @@ test.describe('after searching for kayak', () => {
...
+    await querySummary.load({}, 'no-products');
   });
 
   test('should not display anything', async ({querySummary}) => {
     await expect(querySummary.hydrated).toBeEmpty();
   });
 });
 
-test.describe('when search yields 27 results', () => {
+test.describe('when search yields 27 products', () => {
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/commerce/atomic-commerce-query-summary/e2e/atomic-commerce-query-summary.e2e.ts:37]:
```suggestion
test.describe('when search yields multiple products', () => {
```

File: packages/atomic/src/components/common/load-more/button.tsx [hunk 4]
```diff
@@ -6,20 +6,22 @@ interface LoadMoreButtonProps {
...
+  label,
 }) => {
   if (!moreAvailable) {
     return;
   }
   return (
     <Button
       style="primary"
-      text={i18n.t('load-more-results')}
+      text={i18n.t(label || 'load-more-results')}
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/common/load-more/button.tsx:24]: Why not make the prop mandatory instead?

File: packages/atomic/src/components/common/load-more/summary.tsx [hunk 5]
```diff
@@ -5,19 +5,21 @@ interface LoadMoreSummaryProps {
...
   to,
+  label,
 }) => {
   const wrapHighlight = (content: string) => {
     return `<span class="font-bold text-on-background" part="highlight">${content}</span>`;
   };
 
   const locale = i18n.language;
-  const content = i18n.t('showing-results-of-load-more', {
+  const content = i18n.t(label || 'showing-results-of-load-more', {
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/common/load-more/summary.tsx:22]: Same comment as above

**Review by fbeaudoincoveo — APPROVED** [review]

**Pull request merged**